chall04/edramire.py

```diff
@@ -0,0 +1,52 @@
+#! /usr/bin/env python3
+
+import sys
+import fileinput
+
+
+def process_file(filename):
+    shelfs = []
+    books = 0
+    try:
+        for line in fileinput.input(filename):
+            shelfs_str = line.split(" ")
+            if len(shelfs) == 0:
+                for word in shelfs_str:
+                    word = word.rstrip('\n\r')
+                    if len(word) == 0:
+                        continue
```

Comment:
You're applying `rstrip` on all your words here, but you could have done it earlier on `line` which would have reduced the number of function calls.

I also note that you are jumping empty words, but getting empty words should never happen if you user `.split()` without parameters. Also, using split without parameters will exempt you from using `.rstrip`
```suggestion
        for line in fileinput.input(filename):
            shelfs_str = line.split()
            if len(shelfs) == 0:
                for word in shelfs_str:
```
A little example to illustrate this:
```python
In [1]: "aaaaaaaaaaaaaaaaaaa                 \n\t                  bbbbbbbbbbbbbbbbbbb\n\n\n\n\n\r".split()
Out[1]: ['aaaaaaaaaaaaaaaaaaa', 'bbbbbbbbbbbbbbbbbbb']
```

Then again, you could have replaced this whole loop with a comprehension. Even if that implies catching the `TypeError` exception.